tools/python/create_nifi_reporting_task.py

```diff
@@ -0,0 +1,155 @@
+    task_dict = task.to_dict()
+    return task_dict["component"]["name"]
+
+
+def get_revision_version(task):
+    """Return the ReportingTask revision version"""
+    task_dict = task.to_dict()
+    return task_dict["revision"]["version"]
+
+
+def is_reporting_task_running(task):
+    """Check if the the ReportingTask is already running"""
+    task_dict = task.to_dict()
+    return task_dict["component"]["state"] == "RUNNING"
+
+
+def set_reporting_task_running(task):
+    """Set ReportingTask to RUNNING"""
+    reporting_task_api = nipyapi.nifi.apis.reporting_tasks_api.ReportingTasksApi()
+
+    state = {
+        "revision": {
+            "version": get_revision_version(task)
+        },
+        "disconnected_node_acknowledged": False,
+        "state": "RUNNING"
+    }
+
+    try:
+        return reporting_task_api.update_run_status(id=get_reporting_task_id(task), body=state)
+    except Exception as ex:
+        raise Exception("Could not set ReportingTask [{}] to RUNNING: {}"
+                        .format(get_reporting_task_id(task), str(ex))) from None
+
+
+def main():
+    """Main method with cli argument parsing and ReportingTask logic"""
+    # Construct an argument parser
+    all_args = argparse.ArgumentParser()
+    # Add arguments to the parser
+    all_args.add_argument("-n", "--nifi_api_url", required=True,
+                          help="The NiFi node url to connect to.")
+    all_args.add_argument("-u", "--username", required=True,
+                          help="Username to connect as.")
+    all_args.add_argument("-p", "--password", required=True,
+                          help="Password for the user.")
+    all_args.add_argument("-v", "--nifi_version", required=True,
+                          help="The NiFi product version.")
+    all_args.add_argument("-c", "--cert", required=True,
+                          help="The path to the public certificate.")
+    all_args.add_argument("-m", "--metrics_port", required=True,
+                          help="Metrics port to be set in the ReportingTask.")
+    all_args.add_argument("-t", "--task_name", required=False,
+                          default="StackablePrometheusReportingTask",
+                          help="The name of ReportingTask to create or activate.")
+    args = vars(all_args.parse_args())
+
+    task_name = args["task_name"]
+    port = args["metrics_port"]
+
+    init(args["nifi_api_url"], args["username"], args["password"], args["cert"])
+
+    reporting_task = find_reporting_task(name=task_name, port=port)
+
+    if reporting_task is None:
+        reporting_task = create_reporting_task(name=task_name, port=port,
+                                               version=args["nifi_version"])
+        print(get_reporting_task_name(task=reporting_task) + " [%s] -> CREATED" % reporting_task.id)
+
+    if not is_reporting_task_running(task=reporting_task):
+        reporting_task = set_reporting_task_running(task=reporting_task)
```

Comment:
Shouldn't we catch the exceptions thrown by the methods and log something nice?

Reply:
The exception is the logging here for me.